.github/workflows/release.yaml

```diff
@@ -20,21 +23,124 @@ jobs:
         with:
           go-version: '1.22'
 
+      - name: Build and Package Go executable
+        run: |
+          cd src
+          go mod tidy
+          GOOS=linux GOARCH=amd64 go build -o bootstrap main.go
+          zip -r ../$log_forwarder_zip_file_name .
+          cd ..
+
+      - name: Upload artifact
+        uses: actions/upload-artifact@v3
+        with:
+          name: log-forwarder-zip
+          path: |
+            ${{ env.log_forwarder_zip_file_name }}
+
+      - name: Install AWS SAM CLI
+        run: |
+          pip install aws-sam-cli
+
       - name: Configure AWS Credentials
         uses: aws-actions/configure-aws-credentials@v3
         with:
-          role-to-assume: ${{ secrets.AWS_SAR_PUBLISHER_ROLE }}
+          role-to-assume: ${{ secrets.AWS_S3_PUBLISH_ROLE }}
           aws-region: us-east-2
 
-      - name: Install AWS SAM CLI
+      - name: Build and Package SAM Application and Upload to S3
+        env:
+          bucket_name: unified-lambda-serverless
+          region: us-east-2
         run: |
-          pip install aws-sam-cli
+          BUILD_DIR_BASE=".aws-sam/build"
+          TEMPLATES=(
+            "logging-lambda-metric-polling.yaml"
+            "logging-lambda-metric-stream.yaml"
+            "logging-firehose-metric-polling.yaml"
+            "logging-firehose-metric-stream.yaml"
+            "logging-lambda-firehose-metric-polling.yaml"
+            "logging-lambda-firehose-metric-stream.yaml"
+            "lambda-template.yaml"
+            "logging-lambda-firehose-template.yaml"
+          )
+          
+          for TEMPLATE_FILE in "${TEMPLATES[@]}"; do
+            BASE_NAME=$(basename "$TEMPLATE_FILE" .yaml)
+            BUILD_DIR=".aws-sam/build/$BASE_NAME"
+          
+            sam build --template-file "$TEMPLATE_FILE" --build-dir "$BUILD_DIR"
+            sam package --s3-bucket "$bucket_name" --template-file "$BUILD_DIR/template.yaml" --output-template-file "$BUILD_DIR/$TEMPLATE_FILE"  --region $region
+            if [[ $? -ne 0 ]]; then
+              echo "Error: Failed to package template $TEMPLATE_FILE"
+            fi  
+          
+            aws s3 cp "$BUILD_DIR/$TEMPLATE_FILE" "s3://$bucket_name/$TEMPLATE_FILE"
+            if [[ $? -ne 0 ]]; then
+              echo "Error: Failed to copy $TEMPLATE_FILE to S3 bucket $bucket_name"
+            fi
+          
+          done
+
+  deploy-lambda-code-parallelly-to-all-region:
```

Comment:
Should we run this only when there is change in lambda code? Wondering how can we achieve that?

Reply:
Yes we can do that. I'll move this job to a different file and make it run only when changes are made to src folder?

Reply:
@maya-jha I have added restriction where template upload job runs only if there are changes to any of the templates. And lambda code upload happens only when there's any change to src directory. Let me know what you think

Reply:
We can test when @hrai-nr 's PR is merged.